lib/mongoose_push/service/apns/supervisor.ex

```diff
@@ -0,0 +1,79 @@
+defmodule MongoosePush.Service.APNS.Supervisor do
+  @moduledoc """
+  APNS module supervising Sparrow's PoolSupervisor and APNS State
+  """
+  use Supervisor, id: :apns_supervisor
+  require Logger
+  alias MongoosePush.Application
+
+  @default_endpoints %{
+    dev: "api.development.push.apple.com",
+    prod: "api.push.apple.com"
+}
+
+  @spec start_link([Application.pool_info()]) :: Supervisor.on_start()
+  def start_link(arg) do
+    Supervisor.start_link(__MODULE__, arg)
+  end
+
+  @impl true
+  def init(pool_configs) do
+    sparrow_config = create_sparrow_config(pool_configs)
+
+    children = [
+      Supervisor.child_spec({Sparrow.APNS.Pool.Supervisor, sparrow_config},
+        id: :apns_pool_supervisor
+      ),
+      {MongoosePush.Service.APNS.State, [pool_configs]}
+    ]
+
+    Supervisor.init(children, strategy: :one_for_one)
+  end
+
+  defp create_sparrow_config(pool_configs) do
+    {dev_list, prod_list} = List.foldl(pool_configs, {[], []}, &convert_pool_to_sparrow/2)
+    [{:dev, dev_list}, {:prod, prod_list}]
+  end
+
+  defp convert_pool_to_sparrow({pool_name, pool_config}, {dev_list, prod_list}) do
+    auth_type = {:auth_type, :certificate_based}
+    cert = {:cert, pool_config[:cert]}
+    key = {:key, pool_config[:key]}
+    pool_size = {:worker_num, pool_config[:pool_size]}
+
+    endpoint_mode =
+      case pool_config[:mode] do
```

Comment:
This case seems redundant, as we can just call: 
`@default_endpoints[pool_config[:mode]]`